quantlib/pricingengines/_bondfunctions.pxd

```diff
@@ -0,0 +1,64 @@
+include '../types.pxi'
+
+from quantlib.instruments._bonds cimport Bond as QLBond
+from quantlib.handle cimport shared_ptr
+from quantlib.time._period cimport Frequency
+from quantlib.termstructures._yield_term_structure cimport YieldTermStructure
+from quantlib.time._daycounter cimport DayCounter
+from quantlib.interest_rate cimport InterestRate
+
+cimport quantlib.time._date as _dt
+
+cdef extern from 'ql/cashflows/duration.hpp' namespace 'QuantLib':
+    ctypedef enum Type "QuantLib::Duration::Type":
+        Simple    "QuantLib::Duration::Simple"
+        Macaulay  "QuantLib::Duration::Macaulay"
+        Modified  "QuantLib::Duration::Modified"
+
+cdef extern from 'ql/cashflows/duration.hpp' namespace 'QuantLib':
+    cdef cppclass Duration:
+        Type type
+
+cdef extern from 'ql/pricingengines/bond/bondfunctions.hpp' namespace 'QuantLib':
+    cdef Rate _bf_yield "BondFunctions::yield" (QLBond, Real, DayCounter, int, Frequency, _dt.Date, Real, Size, Rate)
+
+cdef extern from 'ql/pricingengines/bond/bondfunctions.hpp' namespace 'QuantLib':
```

Comment:
You should be able to get rid of the cppclass definition that is overkill compared to the needs. The functions in the struct can be seen as static methods and thus, you should be able (I have not tested that yet) to do this:

```
cdef extern from 'ql/pricingengines/bond/bondfunctions.hpp' namespace 'QuantLib'::BondFunctions':
    cdef _dt.Date startDate(QLBond bond)


There is even the new `staticmethod` keyword in cython that I haven't tested yet (http://docs.cython.org/src/userguide/wrapping_CPlusPlus.html#static-member-method)

I think we should try to make this work as it will drastically simplify the code.